src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/DataContractJsonSerializer.cs

```diff
@@ -497,21 +511,27 @@ internal sealed class DataContractJsonSerializerImpl : XmlObjectSerializer
         private DateTimeFormat? _dateTimeFormat;
         private bool _useSimpleDictionaryFormat;
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
```

Comment:
After further inspection, looks like this suppression is wrong. Even though here we are not passing in knownTypes directly to the serializer, this is not taking into account the fact that the type you pass in may have a KnownTypeAttribute to it which would make it go through the unsafe codePath.